x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/extractor/aggregation/AbstractAggregationDataExtractor.java

```diff
@@ -107,12 +107,11 @@ public long getEndTime() {
         return Optional.ofNullable(processNextBatch());
     }
 
-    private Aggregations search() throws IOException {
+    private Aggregations search() {
         LOGGER.debug("[{}] Executing aggregated search", context.jobId);
         SearchResponse searchResponse = executeSearchRequest(buildSearchRequest(buildBaseSearchSource()));
```

Comment:
Although I see @hendrikmuhs 's point, I prefer the way it's done currently in this PR as it keeps all the settings on the search request together. Even though it means duplicating it, I think it helps readability as it brings all relevant info together when you're reading the `buildSearchRequest` methods.

Reply:
Fair enough, your choice.

To address my concern, what about having an `assert searchRequest.allowPartialSearchResults()`. It's easy to forget otherwise and with the removed response handling its a candidate for a regression if the subclass is re-factored or a new one is introduced.

Reply:
`assert searchRequest.allowPartialSearchResults()` sounds good. I will put that in